### ISSUE TITLE
False-positive: drop unneeded argument

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1607]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | U3.0.10 |
| Language level | PHP 7.4                     |

This inspection should be ignored if internal method uses `func_num_args()`.

Eg.

```php
function where($array, $operator = null, $value = null) {
    if (func_num_args() === 2) {
        $value = $operator;
        $operator = "=";
    }

    return /** ... */;
}
```

Then:

```php
where([], 123) === where([], "=", 123);
```

But when we need that value be really `null`, so we need the argument:

```php
where([], "=", null); // [EA] The argument can be safely dropped, as it's identical to the default value.
```

It is done by Laravel: 

https://github.com/laravel/framework/blob/54324c821e7e08e2b73f5edf279a3230d5629472/src/Illuminate/Support/Traits/EnumeratesValues.php#L891-L938

**Comment by kalessil** [comment]
Fixed!

**Comment by rentalhost** [comment]
@kalessil is this fix already available? It still happens to me.

```
// The null is marked as "[EA] The argument can be safely dropped, as it's identical to the default value."
$laravelCollection->where('review_stars', '!==', null)
```

https://github.com/laravel/framework/blob/8.x/src/Illuminate/Collections/Traits/EnumeratesValues.php#L540-L551
